### DISCUSSION TITLE
Read "hassTokens"

### DISCUSSION THREAD

**Discussion by ifuller1** [shannonhochkins/ha-component-kit#8] [General]
First up thanks for putting this together. It's a great lib and it will save me the misery of working through the shadow dom to update styles!

I am building an app and serving it via home assistants /www/ folder. Ideally the app would read from localStorage to leverage the existing auth token to save a secondary sign in. Is this possible?

P.S. Can we also get non-minified code for core? There were some exceptions that simply output "2" and "5" that were hard to work through.

Cheers!

Ian

**Comment by ifuller1** [comment]
OK I can actually see that you're storing a key with the same name. So I'm wondering why I still get an auth challenge?

**Comment by ifuller1** [comment]
Obviously, that key is actually the key from this library. And it looks like HA negotiates auth over web sockets and it seems a little closed hence why you haven't added this. 

For context, I want to cast my react dashboard to Chrome but it gives me the sign-in page which I can't enter text into.... I wonder how the existing dashboards work...

**Comment by ifuller1** [comment]
This might help: https://www.reddit.com/r/homeassistant/comments/11awl18/bypass_login_page_for_dashboard/

**Comment by shannonhochkins** [comment]
I'll take a look into it and see if we can re-use the tokens, if not i'll provide alternate ways of passing authentication through so we don't have to login on every device!

**Comment by shannonhochkins** [comment]
Hi @ifuller1 - I've released a change to the authentication flow which should solve the token issue, previously i was validating against the current url which has now been removed and it should just read the tokens if they're present in the current domains local storage.

With the casting issue you were having, would you mind providing replicatable steps if it's still an issue?

Cheers!

**Comment by shannonhochkins** [comment]
Going to close this disucussion as it's inactive
